src/main/java/cms/gov/madie/measure/services/TestCaseService.java

```diff
@@ -152,7 +152,7 @@ public TestCase updateTestCase(
     if (measure.getTestCases() == null) {
       measure.setTestCases(new ArrayList<>());
     }
-
+    measureService.verifyAuthorization(username, measureService.findMeasureById(measureId));
```

Comment:
instead of querying the database again, can we reuse the measure object that was fetched on line 146? This would avoid two extra round-trips to the database.